src/parser/statement.js

```diff
@@ -881,10 +884,26 @@ export default class StatementParser extends ExpressionParser {
         if (this.hasPlugin("decorators2")) {
           this.resetStartLocationFromNode(member, decorators[0]);
         }
-        decorators = [];
       }
 
       this.parseClassMember(classBody, member, state);
+      const isClassField = member.kind == null;
+
+      if (isClassField) {
+        if (this.eat(tt.comma)) {
+          this.state.inCommaSeparatedFields = true;
```

Comment:
I don't think we can completely get away without knowing we are within the context of a coma separated production due to ASI and explicit checks for ;.
Im revising it nonetheless.